docs/program/sdks/use-extra-params.md

```diff
@@ -0,0 +1,140 @@
+---
+title: "Using Extra Params with Viam's SDKs"
+linkTitle: "Extra Params"
+weight: 40
+type: "docs"
+description: "Using the extra parameter on resource API methods with Viam's SDKs."
+icon: "/services/img/icons/sdk.svg"
+tags: ["sdk", "extra", "extend"]
+---
+
+How to [utilize](#utilize) and [define](#define) the `extra` parameters that many {{< glossary_tooltip term_id="resource" text="resource" >}} API methods offer in the Go and Python SDKs.
+
+## Utilize
+
+Use these `extra` parameters to pass information to a {{< glossary_tooltip term_id="resource" text="resource's" >}} driver that is not specified as a parameter in the [built-in resource type's API specification](/program/extend/modular-resources/#apis).
+
+To do this, you must code your own modified implementation of the resource type's API for a model.
+See [Extend Viam with Modular Resources](/program/extend/modular-resources/) for more information and [instructions](/program/extend/modular-resources/#use-a-modular-resource-with-your-robot) on modifying API specifications.
+
+An example of how to check the values of keys in an `extra` parameter of a [built-in resource API method](/program/sdks/#add-control-logic), the sensor component's [Readings](/components/sensor/#readings):
```

Comment:
```suggestion
An example of how to check the values of keys in an `extra` parameter of a [built-in resource API method](/program/sdks/#add-control-logic), the sensor component's [Readings](/components/sensor/#readings) method:
```
[opt]

Reply:
Agree sentence is a little awk but I don't think we need to specify method twice!